sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/GetFeedRangeForPartitionKeyValues.scala

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark.udf
+
+import com.azure.cosmos.implementation.SparkBridgeImplementationInternal
+import com.azure.cosmos.spark.CosmosPredicates.requireNotNullOrEmpty
+import org.apache.spark.sql.api.java.UDF2
+
+
+@SerialVersionUID(1L)
+class GetFeedRangeForPartitionKeyValues extends UDF2[String, Object, String] {
+  override def call
+  (
+    partitionKeyDefinitionJson: String,
+    partitionKeyValue: Object
```

Comment:
changing this to array makes sense, but validating it is > 1... this is a new UDF and it is assumed it will only be used for multi partition key. Are you suggesting you prefer a single UDF, and suggesting this as a way of doing that cleanly? Won't it be a breaking change to make this array of objects in that case?

Reply:
I looked at this, it seems Spark UDFs can take only strings as inputs, so it will have to expect json serialized string or an Object. Let me know if there is some other way (I can't find it).

Reply:
https://stackoverflow.com/questions/59026439/how-to-create-a-spark-udf-in-java-which-accepts-array-of-strings

Reply:
I tried this, but when calling the UDF from Spark SQL, where the query text includes a call to a registered UDF, can only seem to pass in strings to the UDF parameters.